pulsar/producer_impl.go

```diff
@@ -177,7 +177,14 @@ func (p *producer) internalCreatePartitionsProducers() error {
 
 	if oldProducers != nil {
 		oldNumPartitions = len(oldProducers)
-		if oldNumPartitions == newNumPartitions {
+		if newNumPartitions < oldNumPartitions {
+			p.log.WithFields(log.Fields{
+				"old_partitions": oldNumPartitions,
+				"new_partitions": newNumPartitions,
+			}).Error("Number of partitions shrank, ignored.")
```

Comment:
Maybe add a little more context in the message something like this? `Unexpected response from get partitions the number of partitions decreased`. 

By ignoring this do you see any potential issues?